### PR TITLE
 Data read/upsert API deserialization failures should include partition and documentId when possible

### PR DESCRIPTION
**Pull request by dhei**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

1. Data read/upsert API deserialization failures should include partition and documentId when possible.
2. Create a new fixture for unit tests to differentiate readonly document from user document.

[AB#61413](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/61413)

### CODE REVIEW

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m [hunk 1]
```diff
@@ -201,8 +201,10 @@ - (void)testReadUserDocumentFromLocalDatabaseNotFound {
 - (void)testUpsertReplacesCorrectlyInAppStorage {
 
   // If
-  MSDocumentWrapper *expectedDocumentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestDocument"]
+  MSDocumentWrapper *expectedDocumentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestAppDocument"]
                                                                            documentType:[MSDictionaryDocument class]
+                                                                              partition:@"readonly"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: You should use the constants for the partition name.

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m [hunk 2]
```diff
@@ -305,8 +307,10 @@ - (void)testUpsertAppDocumentWithValidTTL {
 
   // If
   int ttl = 1;
-  MSDocumentWrapper *expectedDocumentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestDocument"]
+  MSDocumentWrapper *expectedDocumentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestAppDocument"]
                                                                            documentType:[MSDictionaryDocument class]
+                                                                              partition:@"readonly"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: You should use the constants for the partition name.

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m [hunk 3]
```diff
@@ -339,8 +343,10 @@ - (void)testUpsertAppDocumentWithValidTTL {
 - (void)testUpsertAppDocumentWithNoTTL {
 
   // If
-  MSDocumentWrapper *documentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestDocument"]
+  MSDocumentWrapper *documentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestAppDocument"]
                                                                    documentType:[MSDictionaryDocument class]
+                                                                      partition:@"readonly"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: You should use the constants for the partition name.

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m [hunk 4]
```diff
@@ -407,8 +415,10 @@ - (void)testDeleteExistingAppDocument {
 - (void)testDeleteExistingUserDocument {
 
   // If
-  MSDocumentWrapper *documentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestDocument"]
+  MSDocumentWrapper *documentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestUserDocument"]
                                                                    documentType:[MSDictionaryDocument class]
+                                                                      partition:@"user-123"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: You don't need a custom prefixed user partition it's only used for the user table in the storage. These calls are more likely to use `kMSDataUserDocumentsPartition`.

File: AppCenterData/AppCenterDataTests/MSDataTests.m [hunk 5]
```diff
@@ -470,7 +470,7 @@ - (void)testReplaceWithPartitionGoldenPath {
                                  XCTAssertTrue(completionHandlerCalled);
                                  XCTAssertNotNil(actualDocumentWrapper.deserializedValue);
                                  XCTAssertTrue([[actualDocumentWrapper documentId] isEqualToString:@"standalonedocument1"]);
-                                 XCTAssertTrue([[actualDocumentWrapper partition] isEqualToString:@"readonly"]);
+                                 XCTAssertTrue([[actualDocumentWrapper partition] isEqualToString:@"user-123"]);
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDataTests.m]: You don't need a custom prefixed user partition it's only used for the user table in the storage. These calls are more likely to use `kMSDataUserDocumentsPartition`.

File: AppCenterData/AppCenterDataTests/MSDataTests.m [hunk 6]
```diff
@@ -1114,7 +1114,7 @@ - (void)testCreateWithPartitionGoldenPath {
...
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDataTests.m]: You don't need a custom prefixed user partition it's only used for the user table in the storage. These calls are more likely to use `kMSDataUserDocumentsPartition`.

File: AppCenterData/AppCenterDataTests/MSDataTests.m [hunk 7]
```diff
@@ -1896,6 +1896,8 @@ - (void)testReadsFromRemoteIfExpiredAndOnline {
       });
   MSDocumentWrapper *expectedDocumentWrapper = [MSDocumentUtils documentWrapperFromData:testCosmosDbResponse
                                                                            documentType:[MSDictionaryDocument class]
+                                                                              partition:@"user-123"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDataTests.m]: You don't need a custom prefixed user partition it's only used for the user table in the storage. These calls are more likely to use `kMSDataUserDocumentsPartition`.

File: AppCenterData/AppCenterDataTests/MSDataTests.m [hunk 8]
```diff
@@ -1937,13 +1939,17 @@ - (void)testReadsFromRemoteIfNotExpiredAndOnlineWithNoPendingOperation {
   // Mock document in local storage.
   id<MSDocumentStore> localStorageMock = OCMProtocolMock(@protocol(MSDocumentStore));
   self.sut.dataOperationProxy.documentStore = localStorageMock;
-  NSData *jsonFixture = [self jsonFixture:@"validTestDocument"];
+  NSData *jsonFixture = [self jsonFixture:@"validTestUserDocument"];
   MSDocumentWrapper *expectedDocumentWrapper = [MSDocumentUtils documentWrapperFromData:jsonFixture
                                                                            documentType:[MSDictionaryDocument class]
+                                                                              partition:@"user-123"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDataTests.m]: You don't need a custom prefixed user partition it's only used for the user table in the storage. These calls are more likely to use `kMSDataUserDocumentsPartition`.

File: AppCenterData/AppCenterDataTests/MSDataTests.m [hunk 9]
```diff
@@ -1937,13 +1939,17 @@ - (void)testReadsFromRemoteIfNotExpiredAndOnlineWithNoPendingOperation {
...
+                                                                             documentId:@"standalonedocument1"
                                                                         fromDeviceCache:NO];
 
   MSDocumentWrapper *localDocumentWrapper = OCMPartialMock([MSDocumentUtils documentWrapperFromData:jsonFixture
                                                                                        documentType:[MSDictionaryDocument class]
+                                                                                          partition:@"user-123"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDataTests.m]: You don't need a custom prefixed user partition it's only used for the user table in the storage. These calls are more likely to use `kMSDataUserDocumentsPartition`.

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m (hunk 1 above)
Comment (dhei) [reply AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: Done.

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m [hunk 10]
```diff
@@ -382,8 +388,10 @@ - (void)testDeleteAppDocumentForNonExistentDocument {
 - (void)testDeleteExistingAppDocument {
 
   // If
-  MSDocumentWrapper *documentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestDocument"]
+  MSDocumentWrapper *documentWrapper = [MSDocumentUtils documentWrapperFromData:[self jsonFixture:@"validTestAppDocument"]
                                                                    documentType:[MSDictionaryDocument class]
+                                                                      partition:@"user-123"
```
Comment (clpolet) [inline AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: You don't need a custom prefixed user partition it's only used for the user table in the storage. These calls are more likely to use kMSDataUserDocumentsPartition.

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m (hunk 4 above)
Comment (clpolet) [reply AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: I was wrong the returned metadata from the document wrapper should contain the user-account.

File: AppCenterData/AppCenterDataTests/MSDataTests.m (hunk 5 above)
Comment (clpolet) [reply AppCenterData/AppCenterDataTests/MSDataTests.m]: I was wrong the returned metadata from the document wrapper should contain the user-account.

File: AppCenterData/AppCenterDataTests/MSDataTests.m (hunk 6 above)
Comment (clpolet) [reply AppCenterData/AppCenterDataTests/MSDataTests.m]: I was wrong the returned metadata from the document wrapper should contain the user-account.

File: AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m (hunk 10 above)
Comment (clpolet) [reply AppCenterData/AppCenterDataTests/MSDBDocumentStoreTests.m]: I was wrong the returned metadata from the document wrapper should contain the user-account.

File: AppCenterData/AppCenterDataTests/MSDataTests.m (hunk 9 above)
Comment (clpolet) [reply AppCenterData/AppCenterDataTests/MSDataTests.m]: I was wrong the returned metadata from the document wrapper should contain the user-account.

File: AppCenterData/AppCenterDataTests/MSDataTests.m (hunk 8 above)
Comment (clpolet) [reply AppCenterData/AppCenterDataTests/MSDataTests.m]: I was wrong the returned metadata from the document wrapper should contain the user-account.

File: AppCenterData/AppCenterDataTests/MSDataTests.m (hunk 7 above)
Comment (clpolet) [reply AppCenterData/AppCenterDataTests/MSDataTests.m]: I was wrong the returned metadata from the document wrapper should contain the user-account.

**Pull request merged**